src/frontend/mame/ui/menu.cpp

```diff
@@ -321,15 +321,25 @@ void menu::item_append(menu_item item)
 void menu::item_append(menu_item_type type)
 {
 	if (type == menu_item_type::SEPARATOR)
-		item_append(MENU_SEPARATOR_ITEM, nullptr, 0, nullptr, menu_item_type::SEPARATOR);
+		item_append(MENU_SEPARATOR_ITEM, "", 0, nullptr, menu_item_type::SEPARATOR);
 }
 
 //-------------------------------------------------
 //  item_append - append a new item to the
 //  end of the menu
 //-------------------------------------------------
 
-void menu::item_append(const char *text, const char *subtext, UINT32 flags, void *ref, menu_item_type type)
+void menu::item_append(const std::string &text, const std::string &subtext, UINT32 flags, void *ref, menu_item_type type)
+{
+	item_append((std::string&&)text, (std::string&&)subtext, flags, ref, type);
```

Comment:
No, you can't do this - if you're passed a const lvalue reference, the caller doesn't want you to potentially destroy the object out from under them.  The whole point of rvalue references is so that you can consume them destructively if it's advantageous.